### PR TITLE
docs: Clarify on-disk order and size of log entry fields (1.1)

### PR DESCRIPTION
**Pull request by hexagonrecursion**
* This is a backport of https://github.com/borgbackup/borg/pull/6357

### CODE REVIEW

File: docs/internals/data-structures.rst [hunk 1]
```diff
@@ -943,25 +1001,28 @@ See ``borg create --help`` about how to specify the compression level and its de
...
 
-|project_name| uses locks to get (exclusive or shared) access to the cache and
+Borg uses locks to get (exclusive or shared) access to the cache and
 the repository.
 
-The locking system is based on creating a directory `lock.exclusive` (for
-exclusive locks). Inside the lock directory, there is a file indicating
+The locking system is based on renaming a temporary directory
+to `lock.exclusive` (for
+exclusive locks). Inside this directory, there is a file indicating
```
Comment (ThomasWaldmann) [inline docs/internals/data-structures.rst:1009]: it's still os.mkdir in 1.1.x.

**Review by ThomasWaldmann — CHANGES_REQUESTED** [review]
thanks for trying to get 1.1 and 1.2/master better in sync again, but some stuff does not apply to 1.1.x.

**Comment by hexagonrecursion** [comment]
I am confused. I only intended to backport #6357. I am not sure where the rest of the diff came from.

**Comment by ThomasWaldmann** [comment]
Interesting. Do you use `git cherry-pick CHANGESET`?

**Comment by hexagonrecursion** [comment]
Yes. Is there something I do not understand about it?

**Comment by ThomasWaldmann** [comment]
Usually that just works. Or you get some conflicts you have to fix carefully.

**Comment by hexagonrecursion** [comment]
I see what happened. There was a merge conflict. I have looked at it and decided that I want to resolve it by replacing the conflict with the changes from the patch I am backporting. I have followed [this advice](https://stackoverflow.com/a/21777677/14516046) and executed `git checkout --theirs -- docs/`. This did not do what I expected - I expected it to only resolve the conflicts, but it looks like it copied the entire file from 1.2 to 1.1. I am very sorry.

**Review by ThomasWaldmann — APPROVED** [review]
LGTM

**Pull request merged**